ReactWindows/ReactNative/ReactInstanceManager.cs

```diff
@@ -0,0 +1,148 @@
+using ReactNative.Bridge;
+using System.Collections.Generic;
+using System;
+using Windows.UI.Xaml;
+using ReactNative.UIManager;
+using ReactNative.Modules.Core;
+
+namespace ReactNative
+{
+    /// <summary>
+    /// This class is managing instances of <see cref="CatalystInstance" />. It expose a way to configure
+    /// catalyst instance using <see cref="ReactRootView"/> and keeps track of the lifecycle of that
+    /// instance.It also sets up connection between the instance and developers support functionality
+    /// of the framework.
+    ///
+    /// An instance of this manager is required to start JS application in <see cref=" ReactRootView"/>
+    /// #startReactApplication for more info).
+    ///
+    /// The lifecycle of the instance of <see cref="ReactInstanceManager"/> should be bound to the activity
+    /// that owns the <see cref="ReactRootView"/> that is used to render react application using this
+    /// instance manager <see cref=" ReactRootView"/>. It's required to pass
+    /// owning activity's lifecycle events to the instance manager #onPause, #onDestroy, #onResume
+    /// TODO:
+    /// 1.Add lifecycle event hooks
+    /// </summary>
+    public abstract class ReactInstanceManager
```

Comment:
Let's make this an interface and move ReactInstanceManagerImpl -> ReactInstanceManager and hang the builder off that class.